app/src/main/java/com/alphawallet/app/viewmodel/SwapViewModel.java

```diff
@@ -360,6 +400,19 @@ public Web3Transaction buildWeb3Transaction(Quote quote)
         );
     }
 
+    public List<ToolDetails> getTools(Context context)
+    {
+        return new Gson().fromJson(Utils.loadJSONFromAsset(context, "swap_providers_list.json"),
```

Comment:
Duplicated.

Reply:
Did you mean to extract method into another class like `SwapUtils`?

Reply:
Yes, maybe `SwapProviders.get()`.
And is the name `Tool` the same concept as SwapProvider/Exchange?
Should be better to have only one word for one concept.